tests/code/test_minor.py

```diff
@@ -132,26 +134,345 @@ def test_getCommunityEngagementByTerm():
+        neillzEngagements = IndividualRequirement.select()
+        assert neillzEngagements.count() == 5
+        assert neillzEngagements[4].username_id == 'neillz'
+
+        # add a second record for that other student.
+        setCommunityEngagementForUser('add', neillzEngagementData2, 'ramsayb2')
+        neillz2Engagements = IndividualRequirement.select()
+        assert neillz2Engagements.count() == 6
+        assert neillz2Engagements[3].username_id == 'khatts'
+        assert neillz2Engagements[4].username_id == 'neillz'
+        assert neillz2Engagements[4].course == None
+        assert neillz2Engagements[5].username_id == 'neillz'
+        assert neillz2Engagements[5].course == Course.get_by_id(1)
+
+        # Removing requirement
+        # TODO remove nonexisting
+        setCommunityEngagementForUser('remove', khattsEngagementData1, 'ramsayb2')
+        khattsUpdatedEngagements = list(IndividualRequirement.select())
+        assert khattsUpdatedEngagements[0].course == None
+        assert khattsUpdatedEngagements[0].program == Program.get_by_id(9)
+        
+        transaction.rollback()
+
+
+
+@pytest.mark.integration
+def test_getMinorInterest():
+    with mainDB.atomic() as transaction: 
+        # set every users minor interest to no interest
+        User.update(minorInterest = 0).where(User.minorInterest == 1).execute()
+        noStudentsInterested = getMinorInterest()
+        assert noStudentsInterested == []
+
+        # Add a student who has progress towards the minor. They should not be in returned list
+        User.update(minorInterest = 1).where(User.username == 'khatts').execute()
+        minorProgress = getMinorInterest()
+        assert minorProgress == []
+       
+       # Add a student will be returned in the list
+        User.update(minorInterest = 1).where(User.username == 'partont').execute()
+        oneStudentInterested = getMinorInterest()
+        assert len(oneStudentInterested) == 1
+        oneStudentInterested[0]['username'] == 'partont'
+
+        transaction.rollback()
+
+@pytest.mark.integration
+def test_getMinorProgress():
+    with mainDB.atomic() as transaction: 
+        # Make sure the individual requirement table is empty. 
+        IndividualRequirement.delete().execute()
+        noMinorProgress = getMinorProgress()
+
+        assert noMinorProgress == []
+        
+        khattsSustainedEngagement = {"username": "khatts",
+                                     "program": 2,
+                                     "course": None,
+                                     "description": None,
+                                     "term": 3,
+                                     "requirement": 14,
+                                     "addedBy": "ramsayb2",
+                                     "addedOn": "",
+                                     }
+
+        IndividualRequirement.create(**khattsSustainedEngagement)
+        khattsProgress = getMinorProgress()
```

Comment:
Again not the biggest fan of naming variables that are getting **all** of something in a table to be khattsProgress just because we assume she is the only one in there.

Reply:
Can we also make sure that khatts is the one in this list?